app/controllers/partners/base_controller.rb

```diff
@@ -9,7 +9,7 @@ class Partners::BaseController < ApplicationController
   attr_reader :partner
 
   def authenticate!
-    @partner = Partner.find_by_api_key(params[:api_key])
+    @partner = Partner.find_by(api_key: params.permit!.to_h[:api_key])
```

Comment:
We can use `require` here
```ruby
    @partner = Partner.find_by(api_key: params.require(:api_key))
```

or `permit`, better to keep the behaviour
```suggestion
    @partner = Partner.find_by(api_key: params.permit(:api_key)[:api_key])
```

Reply:
@hallelujah done